### ISSUE TITLE
JSON Unmarshalling fails when no content in body

### ISSUE THREAD

**Issue by kierans** [RedCrewOS/api-sdk-creator-js#1]
**Describe the bug**
When receiving a response with no body (eg: an HTTP 204), the JSON unmarshalling fails with an error

`UnhandledPromiseRejectionWarning: Error: Unexpected end of JSON input`

**To Reproduce**

1.  Setup an [HTTP server](https://nodejs.org/en/docs/guides/getting-started-guide/) to return a success response with no body eg:

```
res.statusCode = 200;
res.end();
```

2. Setup a client pipe that uses the `@api-sdk-creator/axios-http-client` `HttpClient` that uses the `jsonUnmarshaller`
 
 ```javascript
(async function run() {
  const client = pipeK(
    createAxiosHttpClient(),
    jsonUnmarshaller()
  );

  await client({
    url: "http://localhost:3000"
  }).toPromise()
}());

```

**Expected behaviour**

HTTP call should execute successfully with no body in the HTTP response.

**Versions:**
 - Node version: v12.21.0
 - `@api-sdk-creator/http-api-client` version: 0.0.1
 - `@api-sdk-creator/axios-http-client` version: 0.1.1 

**Additional context**

**Comment by kierans** [comment]
CC @bfarrelly

**Comment by kierans** [comment]
Investigating this, even with no body in the HTTP response, axios is returning an empty string instead of undefined. So the `jsonUnmarshaller` is trying to parse the JSON. This is a mistake in the axios client as an empty string could still be valid content (depending on the content type).

**Comment by kierans** [comment]
This was fixed in d859e3da9f8d8b3649e47e96ed755febd67b947f and published in `@api-sdk-creator/axios-http-client@0.1.3`

**Comment by kierans** [comment]
I also updated the `jsonUnmarshaller` to respect the content type of the response. If it's not JSON, it will be ignored.
